code/datums/components/catnip.dm

```diff
@@ -0,0 +1,86 @@
+  * meowing incessantly at it as hungry cats tend to do. If the parent is on the ground (well, a turf), the cats will congregate around it, and if the parent is a snack object, they'll begin feasting on it.
+  */
+
+///If a catscan hits prob(catscan_horde_chance), we ignore visibility and multiply the scan range by this. Set to x3 both because it's a reasonable size boost, and also cause x3 looks like a cat face :3c
+#define CATSCAN_HORDE_MULT	3
+///Don't bother if the cat is this far away, just drop them
+#define CATNIP_MAX_DISTANCE	20
+
+/datum/component/catnip
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+
+	///This is the radius we look for cats to entice each scan
+	var/catscan_range = 6
+	///How long we wait to scan for the cats
+	var/catscan_delay = 5 SECONDS
+	///Every catscan, we run prob(this) to see if we should check a much larger radius (x3) for cats
+	var/catscan_horde_chance = 0
+
+	///Which cats are currently following us, as a lazylist
+	var/list/enticed_cats
+	///Cooldown for how long we're waiting to entice cats
+	COOLDOWN_DECLARE(catscan_cooldown)
+
+/datum/component/catnip/Initialize(scan_range = 6, scan_interval = 5 SECONDS, horde_chance = 0, duration = 0)
+	if(!isatom(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	START_PROCESSING(SSdcs, src)
+	catscan_range = scan_range
+	catscan_delay = scan_interval
+	catscan_horde_chance = horde_chance
+	if(duration)
+		QDEL_IN(src, duration)
+
+/datum/component/catnip/Destroy(force, silent)
+	for(var/i in enticed_cats)
+		lose_cat(i)
+	LAZYCLEARLIST(enticed_cats)
+	return ..()
+
+///We're adding a new cat to our harem, hook em up
+/datum/component/catnip/proc/register_cat(mob/living/simple_animal/pet/cat/new_cat)
+	if(!istype(new_cat) || (new_cat in enticed_cats))
+		return
+	LAZYADD(enticed_cats, new_cat)
+	new_cat.current_catnip = parent
+	COOLDOWN_START(new_cat, munchy_frustration, 1 MINUTES)
+	RegisterSignal(new_cat, list(COMSIG_PARENT_QDELETING, COMSIG_MOB_DEATH), .proc/lose_cat)
+
+///A cat is no longer interested in us (or maybe we no longer exist!), so unhook the cat
+/datum/component/catnip/proc/lose_cat(mob/living/simple_animal/pet/cat/bye_cat)
+	UnregisterSignal(bye_cat, list(COMSIG_PARENT_QDELETING, COMSIG_MOB_DEATH))
+	LAZYREMOVE(enticed_cats, bye_cat) // do this before the typecheck just in case it's not a cat anymore
+	if(!istype(bye_cat))
+		return
+	bye_cat.current_catnip = null
+
+/datum/component/catnip/process()
+	if(!COOLDOWN_FINISHED(src, catscan_cooldown))
+		return
+
+	var/list/scannables
+	if(prob(catscan_horde_chance))
+		scannables = range(catscan_range * CATSCAN_HORDE_MULT, get_turf(parent)) //horde scans don't care about visibility, so range instead of hearers
+	else
+		scannables = hearers(catscan_range, get_turf(parent))
+
+	var/mob/living/simple_animal/pet/cat/itercat
+	for(itercat in scannables)
+		if((itercat in enticed_cats) || !COOLDOWN_FINISHED(itercat, munchy_break))
+			continue
+		if(istype(itercat) && !itercat.stat)
```

Comment:
If you substitute `itercat` with a component variable, something like `entice_type = /mob/living/simple_animal/pet/cat`, then this component becomes generic to entice anything: mice with a flute, cats with catnip, you name it

Reply:
Yeah but only cats have the vars and behavior for this, and how they behave was specifically drafted up with cats in mind. If someone wants to repurpose this for rats or dogs or whatever, they'll have to add behavior to said animal for it anyway, so I'll leave it to them for how they want to genericize it

Reply:
components are supposed to be for generic behavior, not for extremely specific one
it's a bad component else

Reply:
The genericness is that the effect can be applied to any kind of atom, which is a step up from before where it was a specific feature of one type of canned food, cat food, and the cat and cat food were both accessing and modifying each other directly. This lets me have catnip cat toys, catnip people, catnip bags of chips, and even catnip ghosts, which is exactly what I want, and is plenty generic in that it's parent agnostic. If someone actually decides that they want to build on this to be able to support other mobs, they'll have a strong base to do so.